shell/platform/darwin/ios/framework/Source/FlutterPlatformViewsController.mm

```diff
@@ -0,0 +1,1080 @@
+// space where the PlatformView is displayed.
+bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
+                                               const SkRect& platformview_boundingrect,
+                                               const SkMatrix& transform_matrix) {
+  SkVector upper_left = clip_rrect.radii(SkRRect::Corner::kUpperLeft_Corner);
+  SkVector upper_right = clip_rrect.radii(SkRRect::Corner::kUpperRight_Corner);
+  SkVector lower_right = clip_rrect.radii(SkRRect::Corner::kLowerRight_Corner);
+  SkVector lower_left = clip_rrect.radii(SkRRect::Corner::kLowerLeft_Corner);
+  SkScalar transformed_upper_left_x = transform_matrix.mapRadius(upper_left.x());
+  SkScalar transformed_upper_left_y = transform_matrix.mapRadius(upper_left.y());
+  SkScalar transformed_upper_right_x = transform_matrix.mapRadius(upper_right.x());
+  SkScalar transformed_upper_right_y = transform_matrix.mapRadius(upper_right.y());
+  SkScalar transformed_lower_right_x = transform_matrix.mapRadius(lower_right.x());
+  SkScalar transformed_lower_right_y = transform_matrix.mapRadius(lower_right.y());
+  SkScalar transformed_lower_left_x = transform_matrix.mapRadius(lower_left.x());
+  SkScalar transformed_lower_left_y = transform_matrix.mapRadius(lower_left.y());
+  SkRect transformed_clip_rect = transform_matrix.mapRect(clip_rrect.rect());
+  SkRRect transformed_rrect;
+  SkVector corners[] = {{transformed_upper_left_x, transformed_upper_left_y},
+                        {transformed_upper_right_x, transformed_upper_right_y},
+                        {transformed_lower_right_x, transformed_lower_right_y},
+                        {transformed_lower_left_x, transformed_lower_left_y}};
+  transformed_rrect.setRectRadii(transformed_clip_rect, corners);
+  return transformed_rrect.contains(platformview_boundingrect);
+}
+
+struct LayerData {
+  SkRect rect;
+  int64_t view_id;
+  int64_t overlay_id;
+  std::shared_ptr<flutter::OverlayLayer> layer;
+};
+
+using LayersMap = std::unordered_map<int64_t, LayerData>;
+
+/// Each of the following structs stores part of the platform view hierarchy according to its
+/// ID.
+///
+/// This data must only be accessed on the platform thread.
+struct PlatformViewData {
+  NSObject<FlutterPlatformView>* view;
+  FlutterTouchInterceptingView* touch_interceptor;
+  UIView* root_view;
+};
+
+}  // namespace
+
+@interface FlutterPlatformViewsController ()
+
+// The pool of reusable view layers. The pool allows to recycle layer in each frame.
+@property(nonatomic, readonly) flutter::OverlayLayerPool* layerPool;
+
+// The platform view's |EmbedderViewSlice| keyed off the view id, which contains any subsequent
+// operation until the next platform view or the end of the last leaf node in the layer tree.
+//
+// The Slices are deleted by the PlatformViewsController.reset().
+@property(nonatomic, readonly)
+    std::unordered_map<int64_t, std::unique_ptr<flutter::EmbedderViewSlice>>& slices;
```

Comment:
Here and a bunch below: notice that the properties here are declared as reference types (`std::unordered_map&`), which allows for `self.slices[foo] = bar` rather than the underlying ivar type `std::unordered_map`, in order to avoid `self.slices` returning a copy.

These should eventually be migrated to Obj-C types like `NSMutableDictionary*`